src/providers/traits.rs

```diff
@@ -0,0 +1,16 @@
+use crate::evaluation::FlattenedContext;
+
+use super::types::{ProviderMetadata, ResolutionDetails};
+
+pub trait FeatureProvider {
+    fn new() -> Self;
+    fn meta_data(&self) -> ProviderMetadata;
+    fn evaluation<T>(
+        &self,
+        flag: String,
+        default_value: T,
+        eval_ctx: FlattenedContext,
+    ) -> (ResolutionDetails<T>, anyhow::Error)
```

Comment:
Same thing, see my comment [here](https://github.com/open-feature/rust-sdk/pull/2/files#r1121067502) on idiomatic Rust `Result`s

Reply:
Updated in next commit